pkg/assets/storage.go

```diff
@@ -136,3 +139,63 @@ func applyCDs(ctx context.Context, cds []string, applier readerApplier, render R
 	}
 	return nil
 }
+
+type volumeSnapshotClassApplier struct {
+	Client dynamic.Interface
+	vc     *unstructured.Unstructured
+}
+
+func (v *volumeSnapshotClassApplier) Reader(objBytes []byte, render RenderFunc, params RenderParams) {
+	var err error
+	if render != nil {
+		objBytes, err = render(objBytes, params)
+		if err != nil {
+			panic(err)
+		}
+	}
+	obj := new(unstructured.Unstructured)
+	err = yaml.Unmarshal(objBytes, &obj)
+	if err != nil {
+		panic(err)
+	}
+
+	v.vc = obj
+}
+
+func (v *volumeSnapshotClassApplier) Applier(ctx context.Context) error {
+	_, _, err := resourceapply.ApplyVolumeSnapshotClass(ctx, v.Client, assetsEventRecorder, v.vc)
+	return err
+}
+
+func ApplyVolumeSnapshotClass(ctx context.Context, kubeconfigPath string, vcs []string, render RenderFunc, params RenderParams) error {
+	applier := &volumeSnapshotClassApplier{}
+	applier.Client = dynamicClient(kubeconfigPath)
+	return applyVolumeSnapshotClass(ctx, applier, vcs, render, params)
+}
+
+func applyVolumeSnapshotClass(ctx context.Context, applier readerApplier, vcs []string, render RenderFunc, params RenderParams) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	for _, vc := range vcs {
+		klog.Infof("Applying volumeSnapshotClass %s", vc)
+		objBytes, err := embedded.Asset(vc)
+		if err != nil {
+			return fmt.Errorf("error getting asset %s: %v", vc, err)
+		}
+		applier.Reader(objBytes, render, params)
+		if err := applier.Applier(ctx); err != nil {
+			klog.Warningf("Failed to apply volumeSnapshotClass api %s: %v", vc, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func dynamicClient(kubeconfigPath string) *dynamic.DynamicClient {
```

Comment:
In the other files in this directory there is a unique function for each client that adds a user agent string to identify that client when it calls the API server. See `appClient` and `coreClient`, for example.